src/webgpu/shader/execution/builtin/abs.spec.ts

```diff
@@ -0,0 +1,301 @@
+export const description = `
+Execution Tests for the 'all' builtin function
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { generateTypes } from '../../types.js';
+
+import { OperandType, runShaderTest } from './builtin.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('abs_float')
+  .uniqueId(0x1f3fc889e2b1727f)
+  .desc(
+    `
+https://www.w3.org/TR/2021/WD-WGSL-20210831#float-builtin-functions
+float abs:
+T is f32 or vecN<f32> abs(e: T ) -> T
+Returns the absolute value of e (e.g. e with a positive sign bit).
+Component-wise when T is a vector.
+(GLSLstd450Fabs)
+`
+  )
+  .params(u =>
+    u
+      .combineWithParams([
+        { storageClass: 'storage', storageMode: 'read_write', access: 'read' },
+      ] as const)
+      .combineWithParams([{ containerType: 'scalar' }, { containerType: 'vector' }] as const)
+      .combineWithParams([{ isAtomic: false }] as const)
+      .combineWithParams([{ baseType: 'f32' }] as const)
```

Comment:
nit:
```suggestion
      .combine('containerType', ['scalar', 'vector'] as const)
      .combine('isAtomic', [false])
      .combine('baseType', ['f32'] as const)
```
(unless there are plans for expansion that would have to change these back to combineWithParams